### PR TITLE
dotnet core installer tool task

### PR DESCRIPTION
**Pull request by bishal-pdMSFT**
Icon TBD

### CODE REVIEW

File: Tasks/DotNetCoreInstaller/task.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
+{
+    "id": "B0CE7256-7898-45D3-9CB5-176B752BFEA6",
+    "name": "DotNetCoreInstaller",
+    "friendlyName": ".Net Core tool Installer (PREVIEW)",
+    "description": "Finds or Downloads and caches specified version spec of .Net Core SDK/runtime and adds it to the PATH.",
```
Comment (sachinma) [inline Tasks/DotNetCoreInstaller/task.json]: Should we have find? Anyway get it reviewed with Atul

File: Tasks/DotNetCoreInstaller/task.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        "DeploymentGroup"
+    ],
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 0,
+        "Minor": 1,
+        "Patch": 0
+    },
+    "preview": true,
+    "satisfies": ["DotNetCore"],
```
Comment (sachinma) [inline Tasks/DotNetCoreInstaller/task.json]: we dont have a demand like that yet, right? should we add it

File: Tasks/DotNetCoreInstaller/dotnetcoreinstaller.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+        let powershellPath = taskLib.which('powershell', true);
+        scriptRunner = taskLib.tool(powershellPath)
+            .line('-NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command')
+            .arg(command);
+
+        primaryUrlSearchString = "dotnet-install: Primary - ";
+        legacyUrlSearchString = "dotnet-install: Legacy - ";
+    } else {
+        let escapedScript = path.join(__dirname, 'externals', 'install-dotnet.sh').replace(/'/g, "''");
```
Comment (sachinma) [inline Tasks/DotNetCoreInstaller/dotnetcoreinstaller.ts]: Why is replace required

File: Tasks/DotNetCoreInstaller/dotnetcoreinstaller.ts [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+    return toolLib.findLocalTool(cachedToolName, version);
+}
+
+async function acquireDotNetCore(packageType: string, version: string): Promise<string> {
+    let downloadUrls = getDownloadUrls(packageType, version);
+    let downloadPath: string;
+
+    try {
+        // try primary url
+        if (!!downloadUrls[0]) {
```
Comment (sachinma) [inline Tasks/DotNetCoreInstaller/dotnetcoreinstaller.ts]: Is there a possibility: If primary url doesn't exist and only secondary exist.

File: Tasks/DotNetCoreInstaller/task.json [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        "Node": {
+            "target": "dotnetcoreinstaller.js"
+        }
+    },
+    "messages": {
+        "ToolFailed": "Tool install failed: %s",
+        "ImplicitVersionNotSupported": "Version should be an explicit version: %s",
+        "PrimaryUrlDownloadFailed": "Failed to download dotnet core package using primary URL. Error: %s.",
+        "LegacyUrlDownloadFailed": "Failed to download dotnet core package using legacy URL as well. Error: %s.",
+        "DownloadFailed": "Failed to download dotnet core package. Please ensure that dotnet core %s of version %s exists. Check out: https://github.com/dotnet/core/blob/master/release-notes/download-archive.md",
```
Comment (sachinma) [inline Tasks/DotNetCoreInstaller/task.json]: Should we have the ensure line given that we have found out the download URL? Or the download URL is only based on patterns and may not exist.

File: Tasks/DotNetCoreInstaller/dotnetcoreinstaller.ts (hunk 4 above)
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreInstaller/dotnetcoreinstaller.ts]: I haven't seen such behavior till now for many versions I tested. we can change if we see this behavior in future.

File: Tasks/DotNetCoreInstaller/task.json (hunk 5 above)
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreInstaller/task.json]: it is based on pattern and may not exist

File: Tasks/DotNetCoreInstaller/dotnetcoreinstaller.ts (hunk 3 above)
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreInstaller/dotnetcoreinstaller.ts]: saw this pattern in NodeTool task and though no harm in using it - they might have seen some issue earlier.

File: Tasks/DotNetCoreInstaller/task.json (hunk 1 above)
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreInstaller/task.json]: will do

File: Tasks/DotNetCoreInstaller/task.json (hunk 2 above)
Comment (bishal-pdMSFT) [inline Tasks/DotNetCoreInstaller/task.json]: There is not existing demand for dotnet core. Talked to kalyan and this name is fine.

**Pull request merged**